src/transformers/models/whisper/generation_whisper.py

```diff
@@ -122,7 +122,9 @@ def _get_attr_from_logit_processors(logits_processor, logit_processor_class, att
     return None
 
 
-def _pad_to_max_length(current_segments, pad_token_id, padding="right", bos_token_tensor=None, cut_off_length=None):
+def _pad_to_max_length(
+    current_segments, pad_token_id, device, padding="right", bos_token_tensor=None, cut_off_length=None
```

Comment:
Do we need to add the device argument? Couldn't we take it from `current_segments`?

Reply:
I thought about it, but there might be some edge cases in which `current_segments` is empty, so I'd rather add a device argument. 
Additionally, `device`  was already passed to `_prepare_decoder_input_ids`, which is a static method that calls `_pad_to_max_length`, so it isn't far-stretched to also pass it to the method IMO!

Reply:
Sounds good to me - thanks for providing the extra context re `_prepare_decoder_input_ids`